src/ansible_creator/subcommands/add.py

```diff
@@ -0,0 +1,167 @@
+    """
+
+    devfile = 0
+
+
+class Add:
+    """Class to handle the add subcommand.
+
+    Attributes:
+        common_resources: List of common resources to copy.
+    """
+
+    common_resources = ("common.devfile",)
+
+    def __init__(
+        self: Add,
+        config: Config,
+    ) -> None:
+        """Initialize the add action.
+
+        Args:
+            config: App configuration object.
+        """
+        self._resource_type: str = config.resource_type
+        self._resource_id: str = self.common_resources[Resources[self._resource_type].value]
+        self._add_path: Path = Path(config.path)
+        self._force = config.force
+        self._overwrite = config.overwrite
+        self._no_overwrite = config.no_overwrite
+        self._creator_version = config.creator_version
+        self._project = config.project
+        self.output: Output = config.output
+        self.templar = Templar()
+
+    def run(self) -> None:
+        """Start scaffolding the resource file."""
+        self._check_add_path()
+        self.output.debug(msg=f"final collection path set to {self._add_path}")
+
+        self._scaffold()
+
+    def _check_add_path(self) -> None:
+        """Validate the provided add path.
+
+        Raises:
+            CreatorError: If the add path does not exist.
+        """
+        if not self._add_path.exists():
+            msg = f"The path {self._add_path} does not exist. Please provide an existing directory."
+            raise CreatorError(msg)
+
+    def unique_name_in_devfile(self) -> str:
+        """Use project specific name in devfile.
+
+        Returns:
+            Unique name entry.
+        """
+        final_name = ".".join(self._add_path.parts[-2:])
+        final_uuid = str(uuid.uuid4())[:8]
+        return f"{final_name}-{final_uuid}"
+
+    def _scaffold(self) -> None:
+        """Scaffold the specified resource file based on the resource type.
+
+        Raises:
+            CreatorError: If unsupported resource type is given.
+        """
+        self.output.debug(f"Started copying {self._project} resource to destination")
+
+        # Call the appropriate scaffolding function based on the resource type
+        if self._resource_type == "devfile":
+            template_data = self._get_devfile_template_data()
+
+        else:
+
+            msg = f"Unsupported resource type: {self._resource_type}"
+            raise CreatorError(msg)
+
+        self._perform_scaffold(template_data)
+
+    def _perform_scaffold(self, template_data: TemplateData) -> None:
+        """Perform the actual scaffolding process using the provided template data.
+
+        Args:
+            template_data: TemplateData
+
+        Raises:
+            CreatorError: If there are conflicts and overwriting is not allowed, or if the
+                      destination directory contains files that will be overwritten.
+        """
+        walker = Walker(
+            resources=self.common_resources,
```

Comment:
Rather than using self.common_resource here we can pass the resource type into this function from the conditional in line 98.  This way when we add devcontainer support we can just pass in devcontainer instead.

Reply:
Hey @cidrblock I have tried to incorporate this change by using resource_type here instead of self.common_resources .. Please review the latest changes made and lemme know if any further changes are required.